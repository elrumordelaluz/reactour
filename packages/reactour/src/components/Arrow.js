import React from 'react'
import styled from '../styled'
import SvgButton from './SvgButton'
import PropTypes from 'prop-types'

const Label = styled.span`
  font-size: 12px;
  line-height: 1;
`

function Arrow({ className, onClick, inverted, label, disabled }) {
  return (
    <SvgButton
      className={className}
      onClick={onClick}
      data-tour-elem={`${inverted ? 'right' : 'left'}-arrow`}
      disabled={disabled}
    >
      {label ? (
        <Label>{label}</Label>
      ) : (
        <svg viewBox="0 0 18.4 14.4">
          <path
            d={
              inverted
                ? 'M17 7.2H1M10.8 1L17 7.2l-6.2 6.2'
                : 'M1.4 7.2h16M7.6 1L1.4 7.2l6.2 6.2'
            }
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeMiterlimit="10"
          />
        </svg>
      )}
    </SvgButton>
  )
}

Arrow.propTypes = {
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  inverted: PropTypes.bool,
  label: PropTypes.node,
  disabled: PropTypes.bool,
}

export default styled(Arrow)`
  color: ${props => (props.disabled ? '#caccce' : '#646464')};

  ${props => (props.inverted ? 'margin-left: 24px;' : 'margin-right: 24px;')};
  ${props =>
    !props.label &&
    `
    width: 16px;
    height: 12px;
    flex: 0 0 16px;
  `};

  &:hover {
    color: ${props => (props.disabled ? '#caccce' : '#000')};
  }
`
