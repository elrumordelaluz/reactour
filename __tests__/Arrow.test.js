import React from 'react';
import Arrow from '../src/components/Arrow.js';
import renderer from 'react-test-renderer';


const arrowDummyProps = {
  className: 'dummy-class',
  onClick: jest.fn(),
  disabled: false,
  inverted: false,
  label: 'dummy label',
};

test('Arrow component must have base methods', () => {
  const component = renderer.create(
    <Arrow
      className={arrowDummyProps.className}
      onClick={arrowDummyProps.onClick}
      disabled={arrowDummyProps.disabled}
      inverted={arrowDummyProps.inverted}
      label={arrowDummyProps.label}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props.onClick).toBe(arrowDummyProps.onClick);

});
