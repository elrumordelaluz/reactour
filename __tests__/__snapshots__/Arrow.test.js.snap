// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component must have base methods 1`] = `
<button
  className="dummy-class sc-htpNat ekWfCE sc-bdVaJa bIMCVx"
  onClick={[Function]}
>
  <span
    className="sc-bwzfXH bWZsjK"
  >
    dummy label
  </span>
</button>
`;
